### PR TITLE
Remove flawed rpois_robust

### PR DESCRIPTION
**Pull request by pbastide**
Major changes : 

- Function `rpois_robust` was flawed (see commit 3d2c3790c60584219bd1891cbdb3b014506b7441), which led to an incorrect handling of heterogeneous lambda values when large values were involved.

- Since `R 4.0`, this function was not needed (now R returns a double instead of a NA for large values of lambda in `rpois`). So I removed the function `rpois_robust`, and used vanilla `rpois` instead.

- DESeq2 can produce NAs in adjusted p values. I added an argument in function `DESeq2.length.createRmd` to replace these NAs with ones, and allow for comparisons with other methods, as explained in the DESeq2 vignette (see doc and messages).

### CODE REVIEW

File: R/generateRmdCodeDiffExp.R [hunk 1]
```diff
@@ -306,8 +309,16 @@ DESeq2.createRmd <- function(data.path, result.path, codefile,
...
                  "DESeq2.ds <- DESeq2.ds.clean"), codefile)
   }
-  writeLines(c(paste("DESeq2.results <- DESeq2::results(DESeq2.ds, independentFiltering = ", independent.filtering, ", cooksCutoff = ", cooks.cutoff, ")", sep = ""),
-               "DESeq2.pvalues <- DESeq2.results$pvalue", 
+  writeLines(paste("DESeq2.results <- DESeq2::results(DESeq2.ds, independentFiltering = ", independent.filtering, ", cooksCutoff = ", cooks.cutoff, ")", sep = ""),
+             codefile) 
+  if (nas.as.ones) {
+    # see https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#i-want-to-benchmark-deseq2-comparing-to-other-de-tools.
+    message("As `nas.as.ones=TRUE`, all NAs in adjusted p values are replaced by 1 to allow for benchmarking with other methods. For more details, see section 'I want to benchmark DESeq2 comparing to other DE tools' from the `DESeq2` vignette (available by running `vignette('DESeq2', package = 'DESeq2')`)")
+    writeLines("DESeq2.length.results$padj <- ifelse(is.na(DESeq2.length.results$padj), 1, DESeq2.length.results$padj)", codefile)
```
Comment (csoneson) [inline R/generateRmdCodeDiffExp.R:317]: DESeq2.length.results -> DESeq2.results

**Comment by csoneson** [comment]
Looks good (see one comment above) - let me know if it's ready to merge.

File: R/generateRmdCodeDiffExp.R (hunk 1 above)
Comment (pbastide) [reply R/generateRmdCodeDiffExp.R]: Good catch, sorry for the bad copy paste.

**Comment by pbastide** [comment]
Thank you for reviewing. If you don't see anything else I think that the PR is good to merge.

**Pull request merged**